benchmarks/Support/Benchmark.php

```diff
@@ -243,16 +249,23 @@ protected function createPredis(): Predis
      */
     public function getBenchmarkMethods(string $filter): array
     {
+        $exclude = null;
+
+        if (! extension_loaded('redis')) {
+            Reporter::printWarning('Skip all PhpRedis tests because redis extension is not loaded');
+            $exclude = 'PhpRedis';
+        }
+
         return array_filter(
             get_class_methods($this),
-            function ($method) use ($filter) {
+            function ($method) use ($exclude, $filter) {
                 if (! str_starts_with($method, 'benchmark')) {
                     return false;
                 }
 
                 $method = substr($method, strlen('benchmark'));
 
-                return ! $filter || preg_match("/$filter/i", strtolower($method));
+                return (! $exclude || $method !== $exclude) && (! $filter || preg_match("/$filter/i", strtolower($method)));
```

Comment:
Can you make this human-readable with a couple of `if` statements?

Reply:
```suggestion
                return ($exclude === false || $method !== $exclude)
                    && ($filter === false || preg_match("/{$filter}/i", strtolower($method)));
```

Reply:
@szepeviktor `$exclude` initialized with `null` so condition your suggested isn't correct and I've already change it to couple of human-readable if statements :smile: